modules/library/xml/src/main/java/org/geotools/xml/filter/FilterCompliancePreProcessor.java

```diff
@@ -855,4 +856,9 @@ public String toString() {
     public boolean requiresPostProcessing() {
         return requiresPostProcessing;
     }
+
+    @Override
+    public Object visit(NativeFilter nativeFilter, Object extraData) {
+        throw new UnsupportedOperationException("Native filter is not supported");
```

Comment:
Seems quite similar to what the default method already offers? But maybe best to keep it, but throw an exception stating that native filters are not encodable in XML (this is what the visitor seem to be doing, prepare filters for XML encoding).